secp256k1_test.go

```diff
@@ -0,0 +1,534 @@
+package secp256k1
+
+import (
+	"bytes"
+	"encoding/hex"
+	"math/big"
+	"math/rand"
+	"testing"
+)
+
+// TODO: Add test vectors. (even though they're included in libsecp itself)
+
+const LoopsN = 150
+
+var Secp256k1Order, _ = new(big.Int).SetString("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141", 16)
+
+func intTo32Bytes(i *big.Int) []byte {
+	res := [32]byte{}
+	serialized := i.Bytes()
+	copy(res[32-len(serialized):], serialized)
+	return res[:]
+}
+
+func fastGeneratePrivateKey(t *testing.T) (key *PrivateKey) {
+	buf := [32]byte{}
+	for {
+		n, err := rand.Read(buf[:])
+		if err != nil || n != len(buf) {
+			t.Fatalf("Failed generating a privatekey %v", err)
+		}
+		privkey, err := ParsePrivateKey(buf)
+		if err == nil {
+			return privkey
+		}
+	}
+}
+
+func TestParseSerializePrivateKey(t *testing.T) {
+	for i := 0; i < LoopsN; i++ {
+		privkey := fastGeneratePrivateKey(t)
+
+		serialized := privkey.Serialize()
+		privkey2, err := ParsePrivateKey(serialized)
+		if err != nil {
+			t.Errorf("Failed parsing privkey %v", err)
+		}
+
+		if *privkey != *privkey2 {
+			t.Errorf("Privkeys aren't equal %v %v", privkey2, privkey)
+		}
+	}
+}
+
+func TestGeneratePrivateKey(t *testing.T) {
+	_, err := GeneratePrivateKey()
+	if err != nil {
+		t.Errorf("Failed generating a privatekey %v", err)
+	}
+}
+
+func TestPrivateKey_Add_fail(t *testing.T) {
+	privkey := fastGeneratePrivateKey(t)
+	privkeyInverse := *privkey
+	privkeyInverse.Negate()
+	err := privkey.Add(privkeyInverse.privkey)
+	if err == nil {
+		t.Errorf("Adding the inverse of itself should fail, %v, %v, %v", err, privkey, privkeyInverse)
```

Comment:
Please annotate the %v's here and everywhere, so that if the test fails the person reading the log wouldn't have to go here to figure out what each one means/